drivers/edac/edac_ibecc.c

```diff
@@ -207,9 +207,11 @@ static int inject_set_param1(const struct device *dev, uint64_t addr)
 	return 0;
 }
 
-static uint64_t inject_get_param1(const struct device *dev)
+static int inject_get_param1(const struct device *dev, uint64_t *value)
```

Comment:
this violates misra dir 4.6: "typedefs that indicate size and signedness should be used in place of the basic numerical types"

Cane we please keep using typedefs instead.

Reply:
@nashif , sorry do not get this. with respect to this change above typedef does not matter. I return here `int` instead of value to be able to return `-ENOTSUP`.

Reply:
If the intent is to return `0` on success and `-errno` on failure then `int` is the appropriate choice, I believe.

Reply:
> If the intent is to return `0` on success and `-errno` on failure then `int` is the appropriate choice, I believe.

directive 4.6 says otherwise though.

Reply:
I guess we might need to state an explicit exception to this rule in our coding style conventions then, since I believe `int` return is used in lots of places of Zephyr, in particular in the core kernel code. If I instead see e.g. `int32_t` I will think that the return value has some other meaning than the usual "0 or -errno", whereas if I see a `int` return value I can assume that it follows the posix error number semantics. I think this is a valuable convention to keep around since that way we keep it separate from some use cases that really do require a fixed-width integer.

Reply:
@nashif I did not understood first that you are talking about this change `int` -> `int32_t`. I thought the comment was about removing some typedefs above. I can change `int` to `int32_t`, although, as @jhedberg mentioned, this is (was?) common for return type in Zephyr API.

Reply:
@nashif Added 617c528  converting `int` to `int32_t`.

Reply:
IMO this line is correct. We should use int (i.e. the natural size of the machine) for return codes, not int32_t or int64_t